### ISSUE TITLE
Simulator not working with capability udid: "auto" for iOS driver

### ISSUE THREAD

**Issue by djonsson** [appium/appium#4000]
**Problem:**
If _udid_ is set to "auto" when running the simulator, you will encounter any of these errors. 

**Error with device attached**

```
org.openqa.selenium.SessionNotCreatedException: 
A new session could not be created. (Original error: Unable to install [ios-build-63.app] to device with id [0a0686fb40f221071f543534543]. 
Error [Error: Command failed: ]) (WARNING: The server did not provide any stacktrace information)
```

**Error with not device attached:**

```
org.openqa.selenium.SessionNotCreatedException: 
A new session could not be created. (Original error: Could not detect udid.) (WARNING: The server did not provide any stacktrace information)
```

**Solution:**
There is no need to actually look for the udid of a device if we are running a simulator test. 

When _udid_ is set to "auto", and if the _app_ capability is an app, the udid should default to an empty string instead, which currently works. As it currently stands, all capabilities except _app_ can be shared between simulator and device testing. It would be nice if _udid_ "auto" also could be shared.

**Comment by mykola-mokhnach** [comment]
The fix has been merged to Appium beta and is available in xcuitest driver since version 2.46.0

**Comment by jlipps** [comment]
🎉  🍾  🎉  issue 4000!
